tempesta_fw/cache.c

```diff
@@ -573,6 +626,8 @@ tfw_cache_add(TfwHttpResp *resp, TfwHttpReq *req)
 
 	if (!cache_cfg.cache || !tfw_cache_msg_cacheable(req))
 		return true;
+	if (!tfw_cache_employ(req, resp))
```

Comment:
Ultimately, it does need the request when making the decision on whether to cache the response or not. Request's Cache-Control arguments that appear pretty soon need to be checked.